django/forms/formsets.py

```diff
@@ -105,6 +105,33 @@ def management_form(self):
             })
         return form
 
+    def _extract_form_indices(self):
+        """Find indices of all submitted forms"""
+        prefix = self.prefix + '-'
+        prefix_len = len(prefix)
+        for key in self.data.keys():
+            if key.startswith(prefix):
+                identifier = key[prefix_len:].split('-', 1)[0]
+                if identifier.isdigit():
+                    yield int(identifier)
+
+    def _extract_total_form_count(self):
+        """Extract the number of submitted forms"""
+        if self.management_form.cleaned_data[TOTAL_FORM_COUNT]:
+            # prefer the management form if data is present
+            return self.management_form.cleaned_data[TOTAL_FORM_COUNT]
+        try:
+            return max(self._extract_form_indices()) + 1
+        except ValueError:  # no forms submitted
+            return 0
+
+    def _extract_initial_form_count(self):
+        """Extract the number of initial forms"""
+        if self.management_form.cleaned_data[INITIAL_FORM_COUNT]:
+            # prefer the management form if data is present
+            return self.management_form.cleaned_data[INITIAL_FORM_COUNT]
+        return 0
```

Comment:
Sorry, none of that was intended as irony or sarcasm, I was trying to understand what you meant about day-to-day use cases.

I think you're right that a hidden field automatically added to each form could replace the initial-form-count portion of the management form.

I've come off too negative on this topic, and I apologize for that. The wording in this pull request and the mailing list discussion sounded to me like a lot of casual conversation about removing and replacing things, without any discussion of why those things exist, what they are used for, and what the issues in replacing them will be (including this line in the PR without any TODO note or commentary about it being incorrect). Probably you understood all of that already, and I just missed out on the key sprint conversations about it.

Please feel free to continue exploring alternative approaches here; I'm looking forward to see what you come up with. I think compatibility with existing dynamic formset JS is a problem, so if we're to do this change as an iterative upgrade, we'll need to find a way to opt-in to the new behavior and preserve the old management-form behavior with a deprecation path.